### PR TITLE
edit borrow position expiry view

### PR DESCRIPTION
**Pull request by graykode**

### CODE REVIEW

File: components/modal/edit-expiry-modal.tsx [hunk 1]
```diff
@@ -1,27 +1,57 @@
...
+    ['coupon-amount-to-edit-expiry', position],
+    () => {
+      return fetchCouponAmountByEpochsBorrowed(
+        position.substitute,
+        position.amount,
+        position.toEpoch.id,
+      )
+    },
+    {
+      enabled: !!position,
```
Comment (onemonster) [inline components/modal/edit-expiry-modal.tsx:33]: Isn't this always true?

File: components/modal/edit-expiry-modal.tsx [hunk 2]
```diff
@@ -1,27 +1,57 @@
...
+  }, [data, selected])
+
+  useEffect(() => {
+    if (expiryEpochIndex > 0) {
+      setSelected(expiryEpochIndex)
+    }
+  }, [expiryEpochIndex, position])
+
   return (
     <Modal show={!!position} onClose={onClose}>
```
Comment (onemonster) [inline components/modal/edit-expiry-modal.tsx:56]: `<Modal show onClose={onClose}>`

File: components/modal/edit-expiry-modal.tsx [hunk 3]
```diff
@@ -47,10 +77,43 @@ const EditExpiryModal = ({
...
       >
-        Confirm
+        {selected === 0
+          ? 'Select expiry date'
+          : expiryEpochIndex === selected
+          ? 'Select expiry date'
+          : (refund === 0n && interest === 0n) || interest > available
+          ? 'Not enough coupons for sale'
+          : interest > balances[position.underlying.address]
+          ? 'Not enough balance'
```
Comment (onemonster) [inline components/modal/edit-expiry-modal.tsx:115]: `Insufficient ${symbol} balance`

File: components/modal/edit-expiry-modal.tsx (hunk 1 above)
Comment (graykode) [reply components/modal/edit-expiry-modal.tsx]: I've cherry picked this commit: https://github.com/clober-dex/coupon.finance/commit/8c99d59df7e34e10f88e316aadc72d4b99c2cd80

File: components/modal/edit-expiry-modal.tsx (hunk 2 above)
Comment (graykode) [reply components/modal/edit-expiry-modal.tsx]: https://github.com/clober-dex/coupon.finance/commit/8c99d59df7e34e10f88e316aadc72d4b99c2cd80

File: components/modal/edit-expiry-modal.tsx (hunk 3 above)
Comment (graykode) [reply components/modal/edit-expiry-modal.tsx]: https://github.com/clober-dex/coupon.finance/commit/3632f5070c8607d105a1087ca662172ccbad4ee7

**Pull request merged**